#! /bin/sh /usr/share/dpatch/dpatch-run
## 02_fix_amd64_s390_build.dpatch by  <danchev@spnet.net>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: fixes amd64 and s390 build - BTS #379620 by Andreas Jochens <aj@andaco.de>

@DPATCH@
diff -urNad bobcat-1.10.0~/string/unescape.cc bobcat-1.10.0/string/unescape.cc
--- bobcat-1.10.0~/string/unescape.cc	2005-12-18 23:16:55.000000000 +0200
+++ bobcat-1.10.0/string/unescape.cc	2006-07-26 12:11:37.000000000 +0300
@@ -7,7 +7,7 @@
 
     unsigned handleOctal(String *dest, String const &src, unsigned pos)
     {
-        unsigned const nOct = 3;                // need exactly 3 octals
+        size_t const nOct = 3;                // need exactly 3 octals
 
         unsigned pos2 = min(pos + nOct, 
                             src.find_first_not_of("01234567", pos));
@@ -28,7 +28,7 @@
 
     unsigned handleHex(String *dest, String const &src, unsigned pos)
     {
-        unsigned const nHex = 2;                // need exactly 2 hex digits
+        size_t const nHex = 2;                // need exactly 2 hex digits
 
         ++pos;                                  // skip the 'x'
         unsigned pos2 = min(pos + nHex, 
